q2_fmt/_peds.py

```diff
@@ -0,0 +1,161 @@
+                           ' IDs where missing references were found:'
+                           ' %s' % (tuple(nan_references),))
+
+    peds_df = _compute_peds(reference_series, table, metadata, time_column,
+                            reference_column, subject_column)
+    return peds_df
+
+
+def _compute_peds(reference_series: pd.Series, table: pd.Series,
+                  metadata: qiime2.Metadata, time_column: str,
+                  reference_column: str,
+                  subject_column: str) -> (pd.DataFrame):
+
+    PEDSserieslist = []
+    for sample in reference_series.index:
+        donor = metadata.loc[sample][reference_column]
+        subject = metadata.loc[sample][subject_column]
+        timepoint = metadata.loc[sample][time_column]
+
+        donorPresentList = _get_observed_features(table, donor)
+        donorNumPresent = _count_observed_features(donorPresentList)
+
+        samplePresentList = _get_observed_features(table, sample)
+
+        intersect = (donorPresentList & samplePresentList).sum()
+        intersectNumPresent = _count_observed_features(intersect)
+
+        peds = (intersectNumPresent/donorNumPresent)
+        (PEDSserieslist.append((sample, peds, intersectNumPresent,
+                                donorNumPresent, donor, subject, timepoint)))
+    PEDSdf = pd.DataFrame(PEDSserieslist,
+                          columns=['id', 'measure',
+                                   'Transfered_Donor_Features',
+                                   'Total_Donor_Features', 'donor', 'subject',
+                                   'group'])
+
+    # use title for correcting ugly names
+    PEDSdf['id'].attrs.update({
+        'title': reference_series.index.name,
+        'description': 'Sample IDS'
+    })
+    PEDSdf['measure'].attrs.update({
+        'title': "PEDS",
+        'description': 'Proportional Engraftment of Donor Strains '
+    })
+    PEDSdf['group'].attrs.update({
+        'title': time_column,
+        'description': 'Time'
+    })
+    PEDSdf["subject"].attrs.update({
+        'title': subject_column,
+        'description': 'ID to link samples across time'
+    })
+    PEDSdf["Transfered_Donor_Features"].attrs.update({
+        'title': "Transfered Donor Features",
+        'description': '...'
+    })
+    PEDSdf['Total_Donor_Features'].attrs.update({
+        'title': "Total Donor Features",
+        'description': '...'
+    })
+    PEDSdf['donor'].attrs.update({
+        'title': reference_column,
+        'description': 'donor'
+    })
+    return PEDSdf
+
+
+def _get_observed_features(table, id):
+    try:
+        present = table.loc[id] > 0
+    except Exception:
+        raise KeyError('There was an error finding the sample %s in'
+                       ' your feature table' % id)
+    return present
+
+
+def _count_observed_features(presentList):
+    numPresent = presentList.sum()
+    return numPresent
+
+    # Hark! These are decomissioned
```

Comment:
Should the helper methods below be removed, or were you saving them for later use?